core/domain/topic_services.py

```diff
@@ -1763,3 +1776,37 @@ def get_chapter_counts_in_topic_summaries(
         })
 
     return topic_chapter_counts_dict
+
+
+def get_published_story_exploration_ids(
+    topic_id: Optional[str] = None
+) -> List[str]:
+    """Returns a list of each exploration id linked to each published chapter.
+    The stories who own the published chapter must also be published. The topic
+    that matches the corresponding topic_id who owns the published stories will
+    have their exploration ids searched.
+
+    If no topic_id is provided, all topics will have their exploration ids
+    searched based on their ownership of published stories and published
+    chapters.
+
+    Args:
+        topic_id: str|None. The id of the topic to search through. When not
+            provided, all topics are searched through.
+
+    Returns:
+        list(str). A list of all exploration ids linked to all the topic(s)'
+        published stories and published chapters.
+    """
+    mappings = ([
+        summary.published_story_exploration_mapping
```

Comment:
Why not just store the .values() of this? 

Optional: I also wonder whether something like itertools.chain() can help you collapse the lists in one go so that you can call list(set()) on the result.

Reply:
The `.values()` was an oversight on my part. My bad.

I finished refactoring this code while using the suggested `itertools.chain()` and `list(set())` on the result.